### ISSUE TITLE
Version 1.17: ZHash not working (version 1.16a is OK)

### ISSUE THREAD

**Issue by UselessGuru** [Lolliedieb/lolMiner-releases#108]
```
lolminer.exe --coin BTG --pool ZHASH.hk.nicehash.com:3369 --user ****************************************.********** --pass x --tls off --log off --apiport 4001 --devicesbypcie --shortstats=2  --longstats=10 --devices 6:0
+---------------------------------------------------------+
|    _       _ __  __ _                   _   _ _____     |
|   | | ___ | |  \/  (_)_ __   ___ _ __  / | / |___  |    |
|   | |/ _ \| | |\/| | | '_ \ / _ \ '__| | | | |  / /     |
|   | | (_) | | |  | | | | | |  __/ |    | |_| | / /      |
|   |_|\___/|_|_|  |_|_|_| |_|\___|_|    |_(_)_|/_/       |
|                                                         |
|              This software is for mining                |
|              Ethash, Etchash                            |
|              Equihash 144/5, 192/7, 210/9               |
|              BeamHash I, II, III                        |
|              ZelHash     (EquihashR 125/4/0)            |
|              Cuck(ar)oo 29                              |
|              Cuckaroo   30 CTX                          |
|              Cuckatoo   31/32                           |
|                                                         |
|                                                         |
|           Made by Lolliedieb, December 2020             |
+---------------------------------------------------------+

Setup Miner...
Device 0:
    Name:    GeForce GTX 1660 SUPER
    Address: 1:0
    Vendor:  NVIDIA Corporation
    Memory:  6144 MByte
    Active:  false (not selected)
Device 1:
    Name:    AMD Radeon RX 5700 XT
    Address: 5:0
    Vendor:  Advanced Micro Devices (AMD)
    Memory:  8176 MByte (8112 MByte free)
    Active:  false (not selected)
Device 2:
    Name:    Radeon RX 580 Series
    Address: 6:0
    Vendor:  Advanced Micro Devices (AMD)
    Memory:  8192 MByte (8138 MByte free)
    Active:  true (Selected Algorithm: Equihash 144/5 AMD)
Device 3:
    Name:    GeForce GTX 750
    Address: 11:0
    Vendor:  NVIDIA Corporation
    Memory:  1024 MByte
    Active:  false (not selected)

Connecting to pool...
Connected to ZHASH.hk.nicehash.com:3369  (TLS disabled)
Subscribed to stratum server
Authorized worker: ****************************************.**********
New target received: 0027ca51790aabf8
New job received: 00000000406d72a4
Start Mining...
Average speed (2s): 3.4 sol/s
Average speed (2s): 28.0 sol/s
Average speed (2s): 16.9 sol/s
Average speed (2s): 19.9 sol/s
Lost connection to stratum server ZHASH.hk.nicehash.com:3369 or server not reachable.
Trying to connect in 1 second
---------------------------------------------
Statistics (15:51:01); Mining: Equihash 144/5
Connected to:
Uptime: 0h 0m 19s
                        GPU 2   Total
Speed (sol/s):           11.1    11.1
Best Share:               0.0     0.0
Shares (A/R):             0/0     0/0
Power (W):               n.a.
Efficiency (sol/s/W):    n.a.    n.a.
Temp (deg C):            n.a.
Fan Speed (%):              0
---------------------------------------------
Average speed (2s): 21.9 sol/s
Connected to ZHASH.hk.nicehash.com:3369  (TLS disabled)
Subscribed to stratum server
Authorized worker: ****************************************.**********
New job received: 00000000406d7a4c
Average speed (2s): 0.0 sol/s
Average speed (2s): 0.0 sol/s
Average speed (2s): 0.0 sol/s
Average speed (2s): 0.0 sol/s
---------------------------------------------
Statistics (15:51:11); Mining: Equihash 144/5
Connected to: ZHASH.hk.nicehash.com:3369
Uptime: 0h 0m 29s
                        GPU 2   Total
Speed (sol/s):            0.0     0.0
Best Share:               0.0     0.0
Shares (A/R):             0/0     0/0
Power (W):               36.9    36.9
Efficiency (sol/s/W):   0.000   0.000
Temp (deg C):              53
Fan Speed (%):             35
---------------------------------------------
Average speed (2s): 0.0 sol/s
Average speed (2s): 0.0 sol/s
```

Loses connection, reconnecs, then stops working (GPU load -> 0).

Version 1.16a is working just fine:

```
lolminer.exe --coin BTG --pool ZHASH.hk.nicehash.com:3369 --user ****************************************.********** --pass x --tls off --log off --apiport 4001 --devicesbypcie --shortstats=2 --longstats=10 --devices 6:0
+---------------------------------------------------------+
|     _       _ __  __ _                   _   _  __      |
|    | | ___ | |  \/  (_)_ __   ___ _ __  / | / |/ /_     |
|    | |/ _ \| | |\/| | | '_ \ / _ \ '__| | | | | '_ \    |
|    | | (_) | | |  | | | | | |  __/ |    | |_| | (_) |   |
|    |_|\___/|_|_|  |_|_|_| |_|\___|_|    |_(_)_|\___/    |
|                                                         |
|              This software is for mining                |
|              Ethash, Etchash                            |
|              Equihash 144/5, 192/7, 210/9               |
|              BeamHash I, II, III                        |
|              ZelHash     (EquihashR 125/4/0)            |
|              Cuck(ar)oo 29                              |
|              Cuckaroo   30 CTX                          |
|              Cuckatoo   31/32                           |
|                                                         |
|                                                         |
|           Made by Lolliedieb, November 2020             |
+---------------------------------------------------------+

Setup Miner...
Device 0:
    Name:    GeForce GTX 1660 SUPER
    Address: 1:0
    Vendor:  NVIDIA Corporation
    Memory:  6144 MByte
    Active:  false (not selected)
Device 1:
    Name:    AMD Radeon RX 5700 XT
    Address: 5:0
    Vendor:  Advanced Micro Devices (AMD)
    Memory:  8176 MByte (8112 MByte free)
    Active:  false (not selected)
Device 2:
    Name:    Radeon RX 580 Series
    Address: 6:0
    Vendor:  Advanced Micro Devices (AMD)
    Memory:  8192 MByte (8138 MByte free)
    Active:  true (Selected Algorithm: Equihash 144/5 AMD)
Device 3:
    Name:    GeForce GTX 750
    Address: 11:0
    Vendor:  NVIDIA Corporation
    Memory:  1024 MByte
    Active:  false (not selected)

Connecting to pool...
Connected to ZHASH.hk.nicehash.com:3369  (TLS disabled)
Subscribed to stratum server
Authorized worker: ****************************************.**********
New target received: 0027ca51790aabf8
New job received: 00000000406d598f
Start Mining...
Average speed (2s): 3.0 sol/s
Average speed (2s): 21.0 sol/s
Average speed (2s): 19.0 sol/s
Average speed (2s): 23.3 sol/s
---------------------------------------------
Statistics (15:49:12); Mining: Equihash 144/5
Connected to: ZHASH.hk.nicehash.com:3369
Uptime: 0h 0m 19s
                        GPU 2   Total
Speed (sol/s):           11.0    11.0
Best Share:               0.0     0.0
Shares (A/R):             0/0     0/0
Power (W):               n.a.
Efficiency (sol/s/W):    n.a.    n.a.
Temp (deg C):            n.a.
Fan Speed (%):              0
---------------------------------------------
Average speed (2s): 25.8 sol/s
Average speed (2s): 21.4 sol/s
Average speed (2s): 21.9 sol/s
Average speed (2s): 24.9 sol/s
Average speed (2s): 26.4 sol/s
---------------------------------------------
Statistics (15:49:22); Mining: Equihash 144/5
Connected to: ZHASH.hk.nicehash.com:3369
Uptime: 0h 0m 29s
                        GPU 2   Total
Speed (sol/s):           23.4    23.4
Best Share:               0.0     0.0
Shares (A/R):             0/0     0/0
Power (W):               96.9    96.9
Efficiency (sol/s/W):   0.240   0.241
Temp (deg C):              55
Fan Speed (%):             35
---------------------------------------------
Average speed (2s): 22.4 sol/s
Average speed (2s): 27.4 sol/s
Average speed (2s): 20.9 sol/s
Average speed (2s): 21.4 sol/s
Average speed (2s): 21.4 sol/s
---------------------------------------------
Statistics (15:49:32); Mining: Equihash 144/5
Connected to: ZHASH.hk.nicehash.com:3369
Uptime: 0h 0m 39s
                        GPU 2   Total
Speed (sol/s):           23.1    23.1
Best Share:               0.0     0.0
Shares (A/R):             0/0     0/0
Power (W):               97.6    97.6
Efficiency (sol/s/W):   0.235   0.236
Temp (deg C):              56
Fan Speed (%):             35
---------------------------------------------
Average speed (2s): 23.5 sol/s
Average speed (2s): 20.9 sol/s
Average speed (2s): 20.9 sol/s
Average speed (2s): 21.4 sol/s
Average speed (2s): 23.4 sol/s
---------------------------------------------
Statistics (15:49:42); Mining: Equihash 144/5
Connected to: ZHASH.hk.nicehash.com:3369
Uptime: 0h 0m 49s
                        GPU 2   Total
Speed (sol/s):           21.9    21.9
Best Share:               0.0     0.0
Shares (A/R):             0/0     0/0
Power (W):               96.4    96.4
Efficiency (sol/s/W):   0.226   0.227
Temp (deg C):              57
Fan Speed (%):             35
---------------------------------------------
Average speed (2s): 24.4 sol/s
GPU 2: Found a share of difficulty 7.9k
GPU 2: Share accepted (80 ms)
Average speed (2s): 26.9 sol/s
Average speed (2s): 26.9 sol/s
Lost connection to stratum server ZHASH.hk.nicehash.com:3369 or server not reachable.
Trying to connect in 1 second
Connected to ZHASH.hk.nicehash.com:3369  (TLS disabled)
Average speed (2s): 12.0 sol/s
Subscribed to stratum server
Authorized worker: ****************************************.**********
New job received: 00000000406d688c
Average speed (2s): 15.9 sol/s
---------------------------------------------
Statistics (15:49:52); Mining: Equihash 144/5
Connected to: ZHASH.hk.nicehash.com:3369
Uptime: 0h 0m 59s
                        GPU 2   Total
Speed (sol/s):           22.2    22.2
Best Share:              7.9K    7.9K
Shares (A/R):             1/0     1/0
Power (W):               85.7    85.7
Efficiency (sol/s/W):   0.257   0.259
Temp (deg C):              56
Fan Speed (%):             35
---------------------------------------------
Average speed (2s): 27.9 sol/s
Average speed (2s): 20.9 sol/s
Average speed (2s): 13.5 sol/s
Average speed (2s): 25.5 sol/s
Average speed (2s): 16.9 sol/s
---------------------------------------------
Statistics (15:50:02); Mining: Equihash 144/5
Connected to: ZHASH.hk.nicehash.com:3369
Uptime: 0h 1m 9s
                        GPU 2   Total
Speed (sol/s):           19.2    19.2
Best Share:              7.9K    7.9K
Shares (A/R):             1/0     1/0
Power (W):               93.8    93.8
Efficiency (sol/s/W):   0.203   0.204
Temp (deg C):              58
Fan Speed (%):             35
---------------------------------------------
Average speed (2s): 19.4 sol/s
Average speed (2s): 21.9 sol/s
Average speed (2s): 22.5 sol/s
Average speed (2s): 26.3 sol/s
Average speed (2s): 20.4 sol/s
---------------------------------------------
Statistics (15:50:12); Mining: Equihash 144/5
Connected to: ZHASH.hk.nicehash.com:3369
Uptime: 0h 1m 19s
                        GPU 2   Total
Speed (sol/s):           22.2    22.2
Best Share:              7.9K    7.9K
Shares (A/R):             1/0     1/0
Power (W):               96.7    96.7
Efficiency (sol/s/W):   0.229   0.230
Temp (deg C):              59
Fan Speed (%):             35
---------------------------------------------
Average speed (2s): 19.9 sol/s
Average speed (2s): 28.5 sol/s
Average speed (2s): 22.9 sol/s
Average speed (2s): 19.9 sol/s
Average speed (2s): 15.0 sol/s
---------------------------------------------
Statistics (15:50:23); Mining: Equihash 144/5
Connected to: ZHASH.hk.nicehash.com:3369
Uptime: 0h 1m 29s
                        GPU 2   Total
Speed (sol/s):           21.5    21.5
Best Share:              7.9K    7.9K
Shares (A/R):             1/0     1/0
Power (W):               97.1    97.1
Efficiency (sol/s/W):   0.220   0.221
Temp (deg C):              60
Fan Speed (%):             35
---------------------------------------------
Average speed (2s): 24.0 sol/s
Average speed (2s): 19.5 sol/s
Average speed (2s): 19.4 sol/s
Average speed (2s): 24.5 sol/s
Average speed (2s): 18.9 sol/s
---------------------------------------------
Statistics (15:50:33); Mining: Equihash 144/5
Connected to: ZHASH.hk.nicehash.com:3369
Uptime: 0h 1m 39s
                        GPU 2   Total
Speed (sol/s):           20.4    20.4
Best Share:              7.9K    7.9K
Shares (A/R):             1/0     1/0
Power (W):               97.9    97.9
Efficiency (sol/s/W):   0.207   0.208
Temp (deg C):              60
Fan Speed (%):             35
---------------------------------------------
Average speed (2s): 16.0 sol/s
Average speed (2s): 20.0 sol/s
Average speed (2s): 23.9 sol/s

```

**Comment by Lolliedieb** [comment]
Thanks for reporting, I will check this :+1:

**Comment by Lolliedieb** [comment]
Fixed in 1.18

**Comment by gocoelho** [comment]
I'm with this problem on version 1.21 of lolminer
